GoldRush.js

```diff
@@ -0,0 +1,320 @@
+        if(player === 1)
+            {
+                let coords =  super.findCoordinate(1)
+                if(coords.x > 0)
+                {
+                    if(this.get(coords.y,coords.x - 1) !== 2)
+                    {
+                        this.matrix[coords.y][coords.x] = '.'
+                        if(this.get(coords.y,coords.x - 1) === 'c')
+                        {
+                            this.player1.score++
+                            this.amount--
+                        }
+                        super.alter(coords.y,coords.x - 1, 1)
+                    }
+             
+                }
+                else
+                {
+                    console.log("This is the leftest you can get to")
+                }
+            }
+            else
+            {
+                let coords =  super.findCoordinate(2)
+                if(coords.x > 0)
+                {
+                    if(this.get(coords.y,coords.x - 1) !== 1)
+                    {
+                        this.matrix[coords.y][coords.x] = '.'
+                        if(this.get(coords.y,coords.x - 1) === 'c')
+                        {
+                            this.player2.score++
+                            this.amount--
+                        }
+                        super.alter(coords.y,coords.x - 1, 2)
+                    }
+                    
+                }
+                else
+                {
+                    console.log("This is the leftest you can get to")
+                }
+        }
+    }
+    moveRight(player)
+    {
+        if(player === 1)
+        {
+            let coords =  super.findCoordinate(1)
+            if(coords.x < this.matrix.length - 1)
+            {
+                if(this.get(coords.y,coords.x + 1) !== 2)
+                {
+                    this.matrix[coords.y][coords.x] = '.'
+                    if(this.get(coords.y,coords.x + 1) === 'c')
+                    {
+                        this.player1.score++
+                        this.amount--
+                    }
+                    super.alter(coords.y,coords.x + 1, 1)
+                }
+            }
+            else
+            {
+                console.log("This is the rightest you can get to")
+            }
+        }
+        else
+        {
+            let coords =  super.findCoordinate(2)
+            if(coords.x < this.matrix.length - 1)
+            {
+                if(this.get(coords.y,coords.x + 1) !== 1)
+                {
+                    this.matrix[coords.y][coords.x] = '.'
+                    if(this.get(coords.y,coords.x + 1) === 'c')
+                    {
+                        this.player2.score++
+                        this.amount--
+                    }
+                    super.alter(coords.y,coords.x + 1, 2)
+                }
+            }
+            else
+            {
+                console.log("This is the rightest you can get to")
+            }
+    }
+    }
```

Comment:
it's hard to read all those `move` functions, and there is a lot of repetition -> better to:
1. deconstruct it into smaller functions (all the `if` statements)
2. make that function receive another parameter of the direction, and use that